### ISSUE TITLE
Reparent to New Node doesn’t use its children’s data for Rect boundaries

### ISSUE THREAD

**Issue by philippedcote** [godotengine/godot#58533]
### Godot version

3.4.2 Stable

### System information

macOS 12.1 (21C52)

### Issue description

When using the command "Reparent to New Node", which essentially is a fancy way of saying "Group to New Node", the newly created Node should use its children Rect data for its own Rect data, as it is the case in all industry standard softwares like Figma, Illustrator and others.

This would avoid many, many unnecessary clicks and headaches.

### Steps to reproduce

1. Add a Node.
2. Change its size and position it elsewhere than the default position.
3. Create another Node.
4. Change its size and position it elsewhere than the default position and the previous node’s position.
5. Select both Nodes.
6. Use the Reparent to New Node command.
7. Notice how the newly created Node completely ignores its children’s Rect data, by using a 40 x 40 Rect Size with a 0, 0 Position.

**Comment by Calinou** [comment]
Have you tested this with both Node2D-based nodes and Control-based nodes? Node2D-based nodes do not have a concept of `rect_size`, only Control-based nodes do.

**Comment by philippedcote** [comment]
> Have you tested this with both Node2D-based nodes and Control-based nodes? Node2D-based nodes do not have a concept of `rect_size`; only Control-based nodes do.

Good point. Yes, it’s for Control Nodes only (I use Godot strictly for UI authoring for the moment). I fixed it in the title. :)

**Comment by philippedcote** [comment]
> unless the existing behavior for Node2D works as expected, in which case this is a bug

I just tried to create 2 Sprites, then used Reparent to New Node using a Position2D Node. Unfortunately, the newly created Node was positioned at 0, 0. :(
